openTSNE/affinity.py

```diff
@@ -255,9 +257,26 @@ def check_perplexity(self, perplexity):
 def build_knn_index(
     data, method, k, metric, metric_params=None, n_jobs=1, random_state=None
 ):
+    if not sp.issparse(data) and metric in [
+        "cosine",
+        "euclidean",
+        "manhattan",
+        "hamming",
+        "dot",
+        "l1",
+        "l2",
+        "taxicab",
+    ]:
+        approx_method = nearest_neighbors.Annoy
```

Comment:
`preferred_approx_method` is more descriptive.

Reply:
done